<?xml version="1.0"?>
<launch>
    <arg name="serial_port" default="/dev/ttyUSB0" />
    <arg name="uwb_port" default="/dev/ttyTHS1" />

    <arg name="drone_id" default="0" />
    <arg name="drone_num" default="2" />
    <!-- <arg name="drone_0_ip" default="192.168.1.1" />
    <arg name="drone_1_ip" default="192.168.1.2" /> -->

    <arg name="drone_0_ip" default="10.0.0.10" />
    <arg name="drone_1_ip" default="10.0.0.11" />
    <arg name="drone_2_ip" default="10.0.0.12" />
    <arg name="drone_3_ip" default="192.168.1.14" />
    <arg name="drone_4_ip" default="192.168.1.14" />
    <arg name="drone_5_ip" default="192.168.1.14" />
    <arg name="ground_station_ip" default="10.0.0.14" />
    <arg name="broadcast_ip" default="10.0.0.255" />
    <arg name="odom_to_broadcast" default="/odom_correction/odom" />

    <node name="goal_tran" type="goal_tran.py" pkg="swarm_bridge" />
    <!-- <node name="odom_correct_init" pkg="status_trans" type="update_odom.py" output="screen">
        <param name="drone_id" value="$(arg drone_id)" />
    </node>
    <node name="status_trans" pkg="status_trans" type="trans.py" output="screen">
        <param name="drone_id" value="$(arg drone_id)" />
        <param name="drone_num" value="$(arg drone_num)" />

    </node> -->
    <!-- <include file="$(find odom_correction)/launch/uwb_correction.launch">
        <arg name="drone_id" value="$(arg drone_id)" />
        <arg name="odom_topic" value="vins_fusion/imu_propagate" />
        <arg name="uwb_topic" value="/nlink_linktrack_nodeframe3" />
    </include> -->
    <include file="$(find nlink_parser)/launch/linktrack.launch">
        <arg name="port_name" value="$(arg uwb_port)" />
    </include>

    <!-- <include file="$(find swarm_bridge)/launch/bridge_udp.launch">
        <arg name="drone_id" value="$(arg drone_id)" />
        <arg name="broadcast_ip" value="$(arg broadcast_ip)" />
    </include> -->

    <node pkg="swarm_bridge" type="bridge_node_uwb" name="bridge_node_uwb" output="screen" respawn="true">
        <param name="self_id" value="$(arg drone_id)" />
        <param name="drone_num" value="$(arg drone_num)" />
        <remap from="~my_odom" to="$(arg odom_to_broadcast)" />
   
    </node>
    
    <!-- <include file="$(find swarm_bridge)/launch/bridge_tcp_drone.launch">
        <arg name="drone_id" default="$(arg drone_id)" />
        <arg name="ground_station_ip" default="$(arg ground_station_ip)" />
        <arg name="drone_num" default="$(arg drone_num)" />
        <arg name="odom_to_broadcast" default="$(arg odom_to_broadcast)" />
        <arg name="drone_0_ip" default="$(arg drone_0_ip)" />
        <arg name="drone_1_ip" default="$(arg drone_1_ip)" />
        <arg name="drone_2_ip" default="$(arg drone_2_ip)" />
        <arg name="drone_3_ip" default="$(arg drone_3_ip)" />
        <arg name="drone_4_ip" default="$(arg drone_4_ip)" />
        <arg name="drone_5_ip" default="$(arg drone_5_ip)" />
    </include> -->
    <!-- <node name="telemetry_air2ground" pkg="swarm_bridge" type="telemetry_air2ground" output="screen">
        <param name="drone_id" value="$(arg drone_id)" />
        <param name="serial_port" value="$(arg serial_port)" />
        <remap from="~all_odom" to="/others_odom" />
    </node> -->
</launch>
